node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -589,3 +697,71 @@ func (o GetDmsPathHelperGeneric) WaitForDmToExist(volumeUuid string, volumeNguid
 	}
 	return "", err
 }
+
+func (o GetDmsPathHelperGeneric) GetDmObjectsByIdentifier(wantedDmObjects []string, outputToCheckIn string) map[string]bool {
+	dmObjects := make(map[string]bool)
+
+	scanner := bufio.NewScanner(strings.NewReader(outputToCheckIn))
+	for scanner.Scan() {
+		identifier, dmObject := o.getLineParts(scanner)
+		if o.IsThisWantedDmObject(wantedDmObjects, identifier) {
+			dmObjects[dmObject] = true
+			logger.Infof("GetDmObjectsByIdentifier: DM Object found: %s for DM Object %s", dmObject, identifier)
+		}
+	}
+
+	return dmObjects
+}
+
+func (GetDmsPathHelperGeneric) getLineParts(scanner *bufio.Scanner) (string, string) {
+	deviceLine := scanner.Text()
+	lineParts := strings.Split(deviceLine, mpathdSeparator)
+	return lineParts[0], lineParts[1]
+}
+
+func (o GetDmsPathHelperGeneric) IsThisWantedDmObject(wantedDmObjects []string, dmObject string) bool {
+	for _, wantedObject := range wantedDmObjects {
+		if strings.Contains(dmObject, wantedObject) {
+			return true
+		}
+	}
+	return false
+}
+
+func (GetDmsPathHelperGeneric) GetFullDmPath(dms map[string]bool, volumeId string) (string, error) {
+	if len(dms) > 1 {
+		return "", &MultipleDmDevicesError{volumeId, dms}
+	}
+
+	var dm string
+	for dm = range dms {
+		break // because its a single value in the map(1 mpath device, if not it should fail above), so just take the first
+	}
+
+	dmPath := filepath.Join(DevPath, filepath.Base(strings.TrimSpace(dm)))
+	return dmPath, nil
+}
+
+func (GetDmsPathHelperGeneric) GetMpathDeviceNameFromProcMountsContent(procMountsFileContent []byte,
+	volumePath string) (string, error) {
+	scanner := bufio.NewScanner(strings.NewReader(string(procMountsFileContent)))
+	for scanner.Scan() {
+		mountLine, lineParts := getLinePartsBySpaces(scanner)
+		if strings.Contains(mountLine, volumePath) {
+			return getMpathDeviceNameFromSplitedLine(lineParts), nil
```

Comment:
why does `lineParts` need to go from `getLinePartsBySpaces` via `GetMpathDeviceNameFromProcMountsContent` to `getMpathDeviceNameFromSplitedLine`? can't it just be computed in `getMpathDeviceNameFromSplitedLine`?

Reply:
and IMO all these 3-lines "get" functions don't add much info, nor make `GetMpathDeviceNameFromProcMountsContent` much shorter (10~ instead of 14~ lines?), and some even seem like code dup (e.g. `getLineParts` vs `getLinePartsBySpaces`)

Reply:
done